docs/DeveloperGuide.md

```diff
@@ -1,34 +1,610 @@
+
+![Sequence Diagram for Repeat Command step 6](./diagrams/repeatstep6.jpg)
+
+##### Step 7: Assign the ArrayList as the RepeatEventList attribute of the original event.
+
+Finally, set the `repeatEventList` using the `setRepeatEventList` command as shown in the following section of the sequence diagram. The results of this process is printed and control returns back to the main program.
+
+![Sequence Diagram for Repeat Command step 7](./diagrams/repeatstep7.jpg)
+
+#### Deadline feature
+
+The user executes ```deadline 1; 7/10/20; 11:20 PM``` command to set the deadline for the 1st event in Personal event list
+to be on the 7th October 2020 at 11:20 PM. 
+The ```deadline``` command calls ```DeadlineCommand#execute()```, adding/updating the personal event deadline. <br>
+Given below is how the deadline command behave: <br>
+
+<p align="center">
+  <img width="414" height="562" src="./diagrams/DeadlineScenario.jpg">
+</p>
+
+The following sequence diagram shows how the deadline operation works: <br>
+
+![Sequence Diagram for Deadline Command](./diagrams/DeadlineSequenceDiagram.jpg)
+
+#### Check feature
+
+The check feature is implemented using the `CheckCommand` class. `CheckCommand` accesses the `Event`s stored within `EventList`s in order to determine if events are occurring within a given time period. It implements the following operations:
+
+- `CheckCommand#getDate(stringDate)` -- Parses a given string to get a LocalDate variable (either the start or end date for the time period).
+- `CheckCommand#getTime(stringTime)` -- Parses a given string to get a LocalTime variable (either the start or end time for the time period).
+- `CheckCommand#checkEventsInTimeRange(eventsList, startDate, endDate, startTime, endTime)` -- Checks each event in the eventsList to see if they occur within the time period defined in the command, and saves all coinciding events in an ArrayList.
+
+These operations are not exposed, and are used as private methods within the `CheckCommand` interface.
+
+Given below is an example usage scenario and how the check feature functions.
+
+Step 1. The user inputs `check 11/5/2020; 5:15 PM; 15/5/2020; 9 PM` in order to check for events occurring between 11th May 2020, 5:15 pm and 15th May 2020, 9:00 pm. This input is received by the Ui, which processes it into a string. The string is parsed by the Parser, which returns a CheckCommand with the date and time parameters stored in it as a string.
+
+Step 2. `CheckCommand#execute()` is called. The command string containing the date and time parameters are split into a String array to separate the different parameters. 
+
+Step 3. Within `CheckCommand#execute()`, `CheckCommand#getDate()` is called to parse the start and end dates, and `CheckCommand#getTime()` is called to parse the start and end times.
+
+This sequence diagram shows how the `getDate` method functions:
+
+![Sequence Diagram for getDate](./diagrams/getDate_seq_diagram.jpg)
```

Comment:
Could this diagram be split into smaller parts? Especially with so many sections of the `alt` frames, it makes the diagram a little difficult to comprehend.